### ISSUE TITLE
Samba sharing issue 

### ISSUE THREAD

**Issue by mgherghi** [MichaIng/DietPi#1893]
Hi all, im having the toughest time mounting my samba share that originates from my merlin modded Asus Ac68U (routers address is 192.168.1.1). My PC finds the shares fine, and without any issue. But once i attempt the same mount in the Pi it fails. I am currently on v6.10 and I am using the drive-manager to add a network drive, but I keep getting this error :
 

│ DietPi-Drive_Manager: mount -t cifs -o username=(edited_out),password=(edited_out)
│ //192.168.1.1/Media /mnt/samba
│ - Exit code: 32
│ - DietPi version: v6.10 | HW_MODEL:3 | HW_ARCH:2 | DISTRO:4
│ - Image creator: DietPi Core Team
│ - Pre-image: Raspbian Lite
│
│ Log file contents:
│ mount error(22): Invalid argument
│ Refer to the mount.cifs(8) manual page (e.g. man mount.cifs)

**Comment by Fourdee** [comment]
@mgherghi 

Thanks for the report 👍 

Does the username/password contain any strange characters (eg: ```$\/:;[]#``` and spaces)?

**Comment by mgherghi** [comment]
no its just letters and numbers

**Comment by Fourdee** [comment]
@mgherghi 

Strange, mounts fine here for Windows share.

Try the following, notice the ```"``` in between the user/password:
```
mkdir -p /mnt/samba
mount -t cifs -o username="CHANGEME",password="CHANGEME" //192.1681.1/Media /mnt/samba
```

**Comment by Fourdee** [comment]
Note to self, ```fstab``` does not support user/password entries contained within either ```"``` or ```'```
```
root@DietPi:~# mount -a
mount error(13): Permission denied
Refer to the mount.cifs(8) manual page (e.g. man mount.cifs)
```

**Comment by mgherghi** [comment]
im getting this error :
mount: can't find /mnt/samba in /etc/fstab

**Comment by Fourdee** [comment]
@mgherghi 

> mount: can't find /mnt/samba in /etc/fstab

Very strange, fstab entry is not required for manual mounting. You received this error, running these commands?
https://github.com/Fourdee/DietPi/issues/1893#issuecomment-402530163

**Comment by mgherghi** [comment]
Yes

**Comment by Fourdee** [comment]
@mgherghi 

Please send us a bug report. We'd like to verify fstab and additional settings.
```
dietpi-bugreport
```

Once completed, reply with the reference code and I will take a look.

**Comment by mgherghi** [comment]
Reference:
b68fb98c-a6b7-4f41-acf5-b2f86b2eaae0

**Comment by mgherghi** [comment]
what should I do?

**Comment by Fourdee** [comment]
@mgherghi 

Thanks for sending the bug report 👍 

I've found the following, the samba server may not support version 2/3 of SMB:
```
[  318.574925] No dialect specified on mount. Default has changed to a more secure dialect, SMB2.1 or later (e.g. SMB3), from CIFS (SMB1). To use the less secure SMB1 dialect to access old servers which do not support SMB3 (or SMB2.1) specify vers=1.0 on mount.
[  318.586740] CIFS VFS: cifs_mount failed w/return code = -22
[  363.406560] systemd-fstab-generator[2105]: Ignoring automount option for root device
[  815.700246] No dialect specified on mount. Default has changed to a more secure dialect, SMB2.1 or later (e.g. SMB3), from CIFS (SMB1). To use the less secure SMB1 dialect to access old servers which do not support SMB3 (or SMB2.1) specify vers=1.0 on mount.
[  815.704388] CIFS VFS: cifs_mount failed w/return code = -22
```

Please try mounting the drive, using SMB1.0:
```
mkdir -p /mnt/samba
mount -t cifs -o username="CHANGEME",password="CHANGEME",vers=1.0 //192.1681.1/Media /mnt/samba
```

**Comment by Fourdee** [comment]
Notes:
- while loop
- vers=3.0
- vers=2.1
- vers=2.0
- vers=1.0

SMB 2.1:
![image](https://user-images.githubusercontent.com/7077026/42382296-c914c3ae-812b-11e8-85b3-19b5d958cd15.png)

SMB 1.0:
![image](https://user-images.githubusercontent.com/7077026/42382334-e26b4aa8-812b-11e8-96fc-2a1d8a2cf15b.png)

**Comment by Fourdee** [comment]
No response, marking as closed.

Should be resolved with v6.12 via SMB version detection during mounting (https://github.com/Fourdee/DietPi/commit/4a9ad692c99981528e8eec11ac113f40693b5592).
